site/_posts/2020-11-15-how-to-securely-implement-oauth-rails.md

```diff
@@ -0,0 +1,262 @@
+---
+layout: blog-post
+title: How to securely implement OAuth in Ruby on Rails 
+description: A detailed overview of securely integrating an Ruby on Rails application with an OAuth provider using the OAuth Authorization Code Grant  
+author: Brett Guy 
+image: blogs/fusionauth-example-angular/oauth-angular-fusionauth.png
+category: blog
+tags: client-ruby
+excerpt_separator: "<!--more-->"
+---
+
+In this tutorial, we'll walk through setting up a basic Ruby on Rails app to securely authenticate with an OAuth2 server using the authorization code grant.
+
+<!--more-->
+
+Many Rails applications traditionally handle authentication, authorization, and user-management within the framework itself. With FusionAuth, we are able to separate those concerns from our application.
+At the same time, we get the benefit of a complete identity solution capable of handling registration, login, MFA, passwordless, SSO, social-logins, and much more.
+
+
+At the end of this tutorial, you will have a working Ruby on Rails application that leverages FusionAuth to authenticate its users.
+
+## Prerequisites
+- Rails 6
+- Docker (optional but preferred for installing FusionAuth)
+
+## Setting up FusionAuth
+In this tutorial we will use Docker for quick setup:
+```bash
+curl -o docker-compose.yml https://raw.githubusercontent.com/FusionAuth/fusionauth-containers/master/docker/fusionauth/docker-compose.yml
+curl -o .env https://raw.githubusercontent.com/FusionAuth/fusionauth-containers/master/docker/fusionauth/.env
+docker-compose up
+```
+
+(Check out the [Download FusionAuth page](https://fusionauth.io/download) for other installation options.)
+
+Once your Docker container is up, it should be running at [localhost:9011](http://localhost:9011). Navigate there and finish the setup steps. 
+When those are completed, login as the administrator that you just created. Now we are ready to rock!
+
+First things first. A [FusionAuth Application](https://fusionauth.io/docs/v1/tech/core-concepts/applications/) is simply something a user can log into.
+As such, we want to create a new application for our Rails app.
+
+### Create an Application
+Select `Applications` from the left nav bar. Then click the green plus sign in the upper-right hand corner to add a new application.
+We will want to configure a few pieces of important information:
+* **Name** (Required): A display name for our application. Enter `fusionauth-rails-app`.
+* **Authorized redirect URL's**: Found under the `Oauth` tab. This URL tells FusionAuth where to redirect to after a user successfully authenticates. 
+* **Logout URL**: Also found under the `Oauth` tab. This URL tells FusionAuth where to redirect to when a user logs out.
+
+{% include _image.liquid src="/assets/img/blogs/fusionauth-example-rails/edit-application.png" alt="Edit Application" class="img-fluid" figure=false %}
```

Comment:
Please have the 'alts' be full sentences.

Reply:
We also do a lot of stuff to make the images of any of the FusionAuth UI consistent. So go ahead and just check in a screenshot of roughly what you want and I'll take care of the consistency. After you've done 2-3 posts, I can walk you through what you need to do. (You're close, though!)